components/netty-http-echo-service/src/main/java/org/wso2/performance/common/netty/echo/EchoHttpServerHandler.java

```diff
@@ -52,28 +53,32 @@
     @Override
     protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) {
         if (h2AggregateContent) {
-            String streamId = getStreamId(request);
-            FullHttpResponse response = EchoHttpServerHandler.buildFullHttpResponse(request);
-            setStreamId(response, streamId);
+            String streamId = request.headers().get(HttpConversionUtil.ExtensionHeaderNames.STREAM_ID.text());
+            FullHttpResponse response = buildFullHttpResponse(request);
+            response.headers().set(HttpConversionUtil.ExtensionHeaderNames.STREAM_ID.text(), streamId);
             ctx.write(response);
```

Comment:
Shall we use writeAndFlush(response) method here too instead of just write() since we don't have a channelReadComplete anymore which did the flushing. Otherwise HTTP/2 might not work properly

Reply:
Yes. I missed that. I will add